src/util/AppContextUtil/Shogun2AppContextUtil.tsx

```diff
@@ -186,23 +186,60 @@ class Shogun2AppContextUtil extends BaseAppContextUtil implements AppContextUtil
         legendUrl
       } = layer.appearance;
 
-      const wmtsTileGrid = new OlTileGridWMTS({
-        origin: layer.source.tileGrid.origin,
-        resolutions: layer.source.tileGrid.resolutions,
-        matrixIds: layer.source.tileGrid.matrixIds
-      });
+      let projection;
+      let wmtsTileGrid;
+      let style;
+
+      if (layer.source.projection) {
+        projection = layer.source.projection;
+      } else {
+        projection = getProjection('EPSG:3857');
+      };
+
+      if (layer.source.tileGrid.origin &&
+        layer.source.tileGrid.resolutions &&
+        layer.source.tileGrid.matrixIds) {
+        wmtsTileGrid = new OlTileGridWMTS({
+          origin: layer.source.tileGrid.origin,
+          resolutions: layer.source.tileGrid.resolutions,
+          matrixIds: layer.source.tileGrid.matrixIds
+        });
+      } else {
+
+        const projectionExtent = projection.getExtent();
+        const size = getWidth(projectionExtent) / 256;
+        const resolutions = new Array(19);
```

Comment:
You assume that always 19 Matrix levels are available in tile grid definition?

Reply:
I got this part from the OpenLayers Examples but you are right, the array could always be different!